javaslang/src/main/java/javaslang/concurrent/Future.java

```diff
@@ -792,6 +791,23 @@ default boolean isFailure() {
     }
 
     /**
+     * Transforms the result of this {@code Future}, whether it is a success or a failure.
+     *
+     * @param f   A transformation
+     * @param <U> Generic type of transformation {@code Try} result
+     * @return A {@code Future} of type {@code U}
+     * @throws NullPointerException if {@code f} is null
+     */
+    default <U> Future<U> transformResult(Function<? super Try<T>, ? extends Try<U>> f) {
```

Comment:
The signature is OK and aligns to existing `transform()` methods that are already present.

I'm think need this instead (all over the lib, in another PR):

```java
<U> Future<U> transformResult(Function<? super Try<? super T>, ? extends Try<? extends U>> f)
```

See also [Sebastian's example](https://github.com/javaslang/javaslang/issues/352#issuecomment-125876580).